server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -85,19 +107,128 @@ public Builder setStored(boolean value) {
 
         @Override
         protected Parameter<?>[] getParameters() {
-            return new Parameter<?>[] { stored, meta };
+            return new Parameter<?>[] { stored, meta, indexOptions };
         }
 
         @Override
         public SparseVectorFieldMapper build(MapperBuilderContext context) {
+            IndexOptions buildIndexOptions = indexOptions.getValue();
+
+            if (buildIndexOptions == null) {
+                buildIndexOptions = new IndexOptions(
+                    true,
+                    new PruningConfig(PruningConfig.DEFAULT_TOKENS_FREQ_RATIO_THRESHOLD, PruningConfig.DEFAULT_TOKENS_WEIGHT_THRESHOLD)
+                );
+            }
+
             return new SparseVectorFieldMapper(
                 leafName(),
-                new SparseVectorFieldType(context.buildFullName(leafName()), stored.getValue(), meta.getValue()),
-                builderParams(this, context)
+                new SparseVectorFieldType(context.buildFullName(leafName()), stored.getValue(), meta.getValue(), buildIndexOptions),
+                builderParams(this, context),
+                buildIndexOptions
             );
         }
     }
 
+    public IndexOptions getIndexOptions() {
+        return this.indexOptions;
+    }
+
+    private static SparseVectorFieldMapper.IndexOptions getDefaultIndexOptions(MappingParserContext context) {
+        if (context.indexVersionCreated().before(SPARSE_VECTOR_PRUNING_INDEX_OPTIONS_VERSION)) {
+            // don't set defaults if this index was created before
+            // we added this functionality in, so it will
+            // not change current index behaviour
+            return null;
```

Comment:
Note: We could technically make the default `prune: false` here instead of null, as that's what the previous behavior is.